apps/language_server/lib/language_server/providers/formatting.ex

```diff
@@ -53,21 +53,49 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
 
   defp can_format?(_uri, _project_dir), do: false
 
-  def should_format?(file_uri, project_dir, inputs) when is_list(inputs) do
+  defp should_format?(file_uri, project_dir, inputs) when is_list(inputs) do
     file_path = file_uri |> SourceFile.abs_path_from_uri()
 
-    inputs
-    |> Stream.flat_map(fn glob ->
-      [
-        Path.join([project_dir, glob]),
-        Path.join([project_dir, "apps", "*", glob])
-      ]
-    end)
-    |> Stream.flat_map(&Path.wildcard(&1, match_dot: true))
-    |> Enum.any?(&(file_path == &1))
+    if File.exists?(file_path) do
+      formatter_dir = find_formatter_dir(project_dir, file_path)
+
+      inputs
+      |> Stream.flat_map(&Path.wildcard(Path.join([formatter_dir, &1]), match_dot: true))
+      |> Enum.any?(&(file_path == &1))
+    else
+      # If the file doesn't actually exist on disk, we can't currently check
+      # whether we should format it via Path.wildcard/2, as that operation is
+      # dependent on the filesystem. So just assume that we should.
```

Comment:
I am not sure if this is necessary in reality?

I needed it to get the formatting test in `server_test.exs` to pass, because that test doesn't create the file on disk that it is formatting.

Previously it passed because `:inputs` was `nil`, so we didn't use this function def. But I've changed that in this commit so as not to have overlapping `.formatter.exs` files (which is documented as undefined behaviour).

An alternative fix would be to do a `File.write!/2` in the test so that we're dealing with a file that actually exists, but I thought that the file-doesn't-exist case might be a real-world use case? But I don't really know.